### ISSUE TITLE
Debugging context lost when send a fixnum to a enum-set-indexer procedure.

### ISSUE THREAD

**Issue by SFurnace** [cisco/ChezScheme#164]
When I was trying to send a fixnum to a procedure created by enum-set-indexer, I got an exception: "invalid memory reference. Some debugging context lost". I know send a number to this procedure is incorrect, but exception won't happen when I send `bignum`, symbols, strings or other objects to this procedure, so it may be a bug?

The version of Chez Scheme is 9.4.1, running on macOS.

```scheme
(define e (make-enumeration '(a b c d)))
((enum-set-indexer e) 1111111111111111111111111111111111111) ;=> #f
((enum-set-indexer e) 1) ;=> error
```

**Comment by burgerrg** [comment]
Nice find! The enum-set-indexer code in enum.ss needs to check that the variable is a symbol before calling symbol-hashtable-ref with it.

**Comment by burgerrg** [comment]
Fixed in f286688b64f877248ab8d8f00528d19363c6ceba
